runtime/shibuya/src/xcm_config.rs

```diff
@@ -66,27 +67,34 @@ pub type CurrencyTransactor = CurrencyAdapter<
     (),
 >;
 
-pub struct AsAssetWithRelay<AssetId, GeneralAssetConverter>(
-    sp_std::marker::PhantomData<(AssetId, GeneralAssetConverter)>,
+pub struct AssetLocationIdConverter<AssetId, AssetMapper>(
+    sp_std::marker::PhantomData<(AssetId, AssetMapper)>,
 );
-impl<AssetId, GeneralAssetConverter> xcm_executor::traits::Convert<MultiLocation, AssetId>
-    for AsAssetWithRelay<AssetId, GeneralAssetConverter>
+impl<AssetId, AssetMapper> xcm_executor::traits::Convert<MultiLocation, AssetId>
+    for AssetLocationIdConverter<AssetId, AssetMapper>
 where
     AssetId: Clone + Eq + Bounded,
-    GeneralAssetConverter: xcm_executor::traits::Convert<MultiLocation, AssetId>,
+    AssetMapper: AssetLocationGetter<AssetId>,
 {
-    fn convert_ref(id: impl Borrow<MultiLocation>) -> Result<AssetId, ()> {
-        if id.borrow().eq(&MultiLocation::parent()) {
+    fn convert_ref(location: impl Borrow<MultiLocation>) -> Result<AssetId, ()> {
+        // TODO should we remove this hardcoding completely?
+        if location.borrow().eq(&MultiLocation::parent()) {
             Ok(AssetId::max_value())
+        } else if let Some(asset_id) = AssetMapper::get_asset_id(location.borrow().clone()) {
+            Ok(asset_id)
         } else {
-            GeneralAssetConverter::convert_ref(id)
+            Err(())
         }
     }
-    fn reverse_ref(what: impl Borrow<AssetId>) -> Result<MultiLocation, ()> {
-        if what.borrow().eq(&AssetId::max_value().into()) {
+
+    fn reverse_ref(id: impl Borrow<AssetId>) -> Result<MultiLocation, ()> {
+        // TODO should we remove this hardcoing completely?
+        if id.borrow().eq(&AssetId::max_value().into()) {
             Ok(MultiLocation::parent())
```

Comment:
Same comment as tin `convert_ref` function.

Reply:
For sure, please polish it